### PR TITLE
UIAUtils.UIATextRangeAttributeValueFetcher.getValue:  catch COMError 

### LINKED ISSUE #7124: eskype for business - keeps giving error sounds with nvda
When using skype for  business with nvda master 14008 skype for business generates error sounds.
When opening the loggs I see some erros:
ERROR - eventHandler.executeEvent (08:45:56):
error executing event: gainFocus on <NVDAObjects.Dynamic_EditableTextWithoutAutoSelectDetectionUIA object at 0x02EDFBF0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 143, in executeEvent
  File "eventHandler.pyc", line 91, in __init__
  File "eventHandler.pyc", line 98, in next
  File "C:\Users\itall101\AppData\Roaming\nvda\addons\remote\globalPlugins\remoteClient\__init__.py", line 310, in event_gainFocus
  File "eventHandler.pyc", line 98, in next
  File "NVDAObjects\__init__.pyc", line 899, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 811, in reportFocus
  File "speech.pyc", line 367, in speakObject
  File "speech.pyc", line 689, in speakTextInfo
  File "NVDAObjects\UIA\__init__.pyc", line 584, in getTextWithFields
  File "NVDAObjects\UIA\__init__.pyc", line 574, in _getTextWithFieldsForUIARange
  File "NVDAObjects\UIA\__init__.pyc", line 409, in _getTextWithFields_text
  File "NVDAObjects\UIA\__init__.pyc", line 213, in _getFormatFieldAtRange
  File "UIAUtils.pyc", line 173, in getValue
Although I am not a software developer, I can't relate the mensioned actions towards skype for business, but I can reproduce this issue and the samme erors will apear in the log file.

### PR DESCRIPTION
**Pull request by michaelDCurran**
IUIAutomationTextRange::getAttributeValue can raise COMError for attribute IDs it doesn't no about (link attribute on E.g. Office 2013 Skype for Business on Windows 7)
Fixes #7124

### CODE REVIEW

File: source/UIAUtils.py [hunk 1]
```diff
@@ -170,7 +170,11 @@ def __init__(self,textRange):
 		self.textRange=textRange
 
 	def getValue(self,ID,ignoreMixedValues=False):
-		val=self.textRange.getAttributeValue(ID)
+		try:
+			val=self.textRange.getAttributeValue(ID)
+		except COMError:
+			# #7124: some text attributes are not supported in  older Operating Systems
```
Comment (feerrenrut) [inline source/UIAUtils.py]: Is this something that would be helpful to log? Would that cause a lot of noise in the log? I'm considering a case where we are investigating why something does not work in a particular OS for instance.

What happens when `reservedNotSupportedValue` is returned, is this handled gracefully in all cases?

**Comment by michaelDCurran** [comment]
This method is called a great deal. I feel it would be pretty  noisy for 
the log yes. UIA implementations can choose to support any of the UIA 
text attributes. In theory they should be returning 
ReservedNotSupportedValue themselves, but it seems that they only do 
this for ones they know but do not support, so if a new attribute is 
added to the spec, most likely an earlier OS just throws the exception.
Re it being handled gracefully: See 
```source/NVDAObjects/UIA/__init__.py``` Line 214 for  an example.

**Review by feerrenrut — APPROVED** [review]
Ok, thanks!

**Comment by jcsteh** [comment]
@MichaelDCurran, @feerrenrut and I all agreed this should be merged into master before the translation freeze despite it not being fully incubated, as it it is a p1 fix which needs to be included in the release.

**Pull request merged**